<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Version 0.1 • bcbioSingleCell</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



  <link href="../extra.css" rel="stylesheet">
  

<meta property="og:title" content="Version 0.1" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-news">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">bcbioSingleCell</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.4.14</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/hbc/bcbioSingleCell/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
      <h1 data-toc-skip>Changelog <small>0.1</small></h1>
      
    </div>

    <div id="bcbiosinglecell-017-2018-05-15" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiosinglecell-017-2018-05-15" class="anchor"></a>bcbioSingleCell 0.1.7 (2018-05-15)</h2>
<div id="new-functions-2" class="section level3">
<h3 class="hasAnchor">
<a href="#new-functions-2" class="anchor"></a>New functions</h3>
<ul>
<li>Added support for Uniform Manifold Approximation and Projection (UMAP) with the <code>plotUMAP</code> and <code>fetchUMAPData</code> functions. These work similarly to the other <code>plotDimensionalReduction</code> and <code>fetchData</code> functions.</li>
</ul>
</div>
<div id="major-changes-18" class="section level3">
<h3 class="hasAnchor">
<a href="#major-changes-18" class="anchor"></a>Major changes</h3>
<ul>
<li>Now adding sample-level metadata into <code>colData</code> slot, for better downstream compatibility with other packages that work with <code>SingleCellExperiment</code> container class. Unique per-sample rows are still saved internally in the <code>sampleData</code> slot.</li>
<li>Now recommending ECDF as the default geom for quality control plots, where applicable.</li>
<li>
<code>filterCells</code> now supports <code>minUMIs = c("knee", "inflection")</code> for automatic filtering based on the cellular barcode ranks. Internally this is handled by <code><a href="https://rdrr.io/pkg/DropletUtils/man/barcodeRanks.html">DropletUtils::barcodeRanks</a></code>.</li>
</ul>
</div>
<div id="minor-changes-46" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-46" class="anchor"></a>Minor changes</h3>
<ul>
<li>Attempting to re-enable <code>libgsl-dev</code> installation for <a href="https://bioconductor.org/packages/zinbwave/">zinbwave</a> on <a href="https://travis-ci.org/">Travis CI</a>.</li>
<li>Suggesting <a href="https://bioconductor.org/packages/BiocParallel/">BiocParallel</a> for <a href="https://bioconductor.org/packages/zinbwave/">zinbwave</a> call in <code>diffExp</code>.</li>
<li>Now importing <a href="http://satijalab.org/seurat/">Seurat</a> functions into NAMESPACE.</li>
<li>Consolidated <code>fetchData</code> functions in the documentation.</li>
<li>Consolidated <code>plotDimensionalReduction</code> functions in the documentation.</li>
<li>Updated <code>aggregateReplicates</code> internal code. This function again only supports aggregation of <code>bcbioSingleCell</code> objects that have been filtered using the <code>filterCells</code> function.</li>
<li>Now using <code>Seurat::Convert</code> internally to coerce <code>seurat</code> class object to <code>SingleCellExperiment</code>, using <code>as(seurat, "SingleCellExperiment")</code>. This utility function was added to <a href="http://satijalab.org/seurat/">Seurat</a> v2.3.1.</li>
</ul>
</div>
</div>
    <div id="bcbiosinglecell-018-2018-05-16" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiosinglecell-018-2018-05-16" class="anchor"></a>bcbioSingleCell 0.1.8 (2018-05-16)</h2>
<div id="minor-changes-45" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-45" class="anchor"></a>Minor changes</h3>
<ul>
<li>Initial <code>updateObject</code> method support for <code>bcbioSingleCell</code> class.</li>
<li>Relaxed <code>validObject</code> validity check to not require sample-level metadata in <code>colData</code> yet.</li>
</ul>
</div>
</div>
    <div id="bcbiosinglecell-019-2018-05-18" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiosinglecell-019-2018-05-18" class="anchor"></a>bcbioSingleCell 0.1.9 (2018-05-18)</h2>
<div id="major-changes-17" class="section level3">
<h3 class="hasAnchor">
<a href="#major-changes-17" class="anchor"></a>Major changes</h3>
<ul>
<li>No longer attempting to sanitize the rownames for <code>seurat</code> objects in coercion method. This helps maintain the gene symbol appearance in plotting functions for genes with hyphens in the names.</li>
</ul>
</div>
<div id="minor-changes-44" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-44" class="anchor"></a>Minor changes</h3>
<ul>
<li>Using <code><a href="https://rdrr.io/pkg/BiocParallel/man/SerialParam-class.html">BiocParallel::SerialParam</a></code> internally for zinbwave in <code>diffExp</code>.</li>
<li>Simplified <code>cell2sample</code> internal code to always use <code>mapCellsToSamples</code> instead of attempting to use a stashed <code>vector</code> inside <code>metadata</code> for <code>SingleCellExperiment</code> method.</li>
<li>Removed internal <code>.applyFilterCutoffs</code>, which is no longer necessary since this functionality is supported in the S4 subset method.</li>
<li>Simplified assert checks inside <code>fetchGene</code> functions.</li>
<li>
<code>plotCellTypesPerCluster</code>: revert back to <code>dark = TRUE</code> by default.</li>
<li>Consolidated <code>plotMarker</code> and <code>plotFeature</code> functions in the documentation.</li>
<li>
<code>sanitizeMarkers</code>: Improved gene identifier matching.</li>
<li>
<code>topMarkers</code> now defaults to <code>coding = FALSE</code> by default, since not all datasets will contain biotype information.</li>
</ul>
</div>
</div>
    <div id="bcbiosinglecell-0110-2018-05-19" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiosinglecell-0110-2018-05-19" class="anchor"></a>bcbioSingleCell 0.1.10 (2018-05-19)</h2>
<div id="minor-changes-43" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-43" class="anchor"></a>Minor changes</h3>
<ul>
<li>
<code>plotCellTypesPerCluster</code> is using <code>dark = TRUE</code> by default again.</li>
<li>Fixed <code>cell2sample</code> handling for multiplexed <a href="https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/what-is-cell-ranger">Cell Ranger</a> data loaded up with <code>readCellRanger</code>. Need to use stashed <code>cell2sample</code> factor saved in <code>metadata</code>, rather than attempting to calculate on the fly with <code>mapCellsToSamples</code>.</li>
<li>Updated <a href="https://travis-ci.org/">Travis CI</a> build checks to include bioc-release on <a href="https://www.apple.com/macos/">macOS</a>.</li>
</ul>
</div>
</div>
    <div id="bcbiosinglecell-0111-2018-05-23" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiosinglecell-0111-2018-05-23" class="anchor"></a>bcbioSingleCell 0.1.11 (2018-05-23)</h2>
<div id="new-functions-1" class="section level3">
<h3 class="hasAnchor">
<a href="#new-functions-1" class="anchor"></a>New functions</h3>
<ul>
<li>UMAP is now supported. This functionality is provided in: <code>plotUMAP</code>, <code>plotMarkerUMAP</code>, and <code>plotFeatureUMAP</code>. Corresponding fetch functions, <code>fetchUMAPData</code> and <code>fetchUMAPExpressionData</code>, have also been added.</li>
<li>
<code>plotGene</code>: Added <code>seurat</code> method support. If advanced customization of the plot is needed, use <code>plotDot</code> or <code>plotViolin</code> instead, or refer to the <a href="http://satijalab.org/seurat/">Seurat</a> documentation for alternates.</li>
</ul>
</div>
<div id="major-changes-16" class="section level3">
<h3 class="hasAnchor">
<a href="#major-changes-16" class="anchor"></a>Major changes</h3>
<ul>
<li>Dimensional reduction and marker plots no longer use dark mode by default. The default color palette support for marker plots has been improved to consistently use <a href="https://cran.r-project.org/package=viridis">viridis</a>.</li>
<li>
<code>diffExp</code>: improved internal code to work directly on <code>SingleCellExperiment</code>, removing the need to pass <code>design</code> and <code>group</code> parameters internally. Also added unit testing against <a href="https://bioconductor.org/packages/zinbwave/">zinbwave</a>, <a href="https://github.com/statOmics/zingeR/">zingeR</a>, and <a href="https://bioconductor.org/packages/edgeR/">edgeR</a> support. <a href="https://bioconductor.org/packages/DESeq2/">DESeq2</a> is supported but runs slowly.</li>
<li>Reworked <code>plotFeature</code> and <code>plotMarker</code> family of functions. Improved the color palette support when <code>dark = FALSE</code>, now using a flipped <a href="https://cran.r-project.org/package=viridis">viridis</a> plasma color palette.</li>
<li>
<code>aggregateReplicates</code> function has been reworked to return a <code>SingleCellExperiment</code> object instead of <code>bcbioSingleCell</code>. The v0.2.4 update of <a href="https://bioinformatics.sph.harvard.edu/bcbioRNASeq/">bcbioRNASeq</a> behaves similarly with this generic.</li>
</ul>
</div>
<div id="minor-changes-42" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-42" class="anchor"></a>Minor changes</h3>
<ul>
<li>Reworked the internal handling of some <code>seurat</code> <code>SingleCellExperiment</code> method support, using <code>as(x, "SingleCellExperiment")</code> internally, which uses the new <code>Seurat::Convert</code> function.</li>
<li>Made some previously deprecated functions now defunct: <code>plotClusters</code>, <code>plotTSNEExpressionData</code>, <code>loadSingleCellRun</code>, <code>darkTheme</code>, <code>pcCutoff</code>, <code>quantileHeatmap</code>, <code>plotKnownMarkers</code>, <code>readMarkers</code>, <code>readMarkersFile</code>.</li>
<li>Made <code>plotFeatures</code>, <code>plotMarker</code>, and <code>plotMarkers</code> functions defunct.</li>
<li>
<code>plotPCElbow</code> now returns a plot grid.</li>
<li>
<code>sanitizeMarkers</code>: improved internal code for supported bcbio stashed metadata, including <code>rowRanges</code>.</li>
</ul>
</div>
</div>
    <div id="bcbiosinglecell-0112-2018-05-25" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiosinglecell-0112-2018-05-25" class="anchor"></a>bcbioSingleCell 0.1.12 (2018-05-25)</h2>
<div id="major-changes-15" class="section level3">
<h3 class="hasAnchor">
<a href="#major-changes-15" class="anchor"></a>Major changes</h3>
<ul>
<li>No longer using automatic camel case sanitization for <code>metrics</code> or <code>fetchData</code> return column names.</li>
<li>Improved <a href="http://rmarkdown.rstudio.com/">R Markdown</a> clustering and marker templates to optionally support UMAP and dark mode in the YAML parameters.</li>
</ul>
</div>
<div id="minor-changes-41" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-41" class="anchor"></a>Minor changes</h3>
<ul>
<li>Using original <a href="http://satijalab.org/seurat/">Seurat</a> mapping names for data: tSNE_1, tSNE_2, PC1, PC2, UMAP1, UMAP2.</li>
<li>Ensure transcript-level counts always have <code>stripTranscriptVersions</code> command applied, to remove the <a href="https://www.ensembl.org/">Ensembl</a> transcript versions if present.</li>
<li>No longer using labels (e.g. A, B, C, D) on <code>ggplot</code> grid return.</li>
<li>Note that “phase” has been renamed to “Phase” in the <a href="http://rmarkdown.rstudio.com/">R Markdown</a> clustering for cell-cycle regression PCA.</li>
</ul>
</div>
</div>
    <div id="bcbiosinglecell-0113-2018-05-29" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiosinglecell-0113-2018-05-29" class="anchor"></a>bcbioSingleCell 0.1.13 (2018-05-29)</h2>
<div id="minor-changes-40" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-40" class="anchor"></a>Minor changes</h3>
<ul>
<li>t-SNE and UMAP plot improvements.</li>
</ul>
</div>
</div>
    <div id="bcbiosinglecell-0114-2018-06-05" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiosinglecell-0114-2018-06-05" class="anchor"></a>bcbioSingleCell 0.1.14 (2018-06-05)</h2>
<div id="major-changes-14" class="section level3">
<h3 class="hasAnchor">
<a href="#major-changes-14" class="anchor"></a>Major changes</h3>
<ul>
<li>Multiplexed <a href="https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/what-is-cell-ranger">Cell Ranger</a> barcodes are now reformatted to be more readable and compatible with the <code>mapCellsToSamples</code> utility function. This change helps simplify the internal code for <code>cell2sample</code>. For example in the pbmc4k dataset, the barcode IDs are sanitized from <code>TTTGGTTTCGCTAGCG-1</code> to “<code>pbmc4k_1_TTTGGTTTCGCTAGCG</code>”. The “<code>1</code>” here denotes 1 sample in the matrix, which is how <a href="https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/what-is-cell-ranger">Cell Ranger</a> denotes multiplexed samples in a single counts matrix. Note that the “<code><a href="https://rdrr.io/r/base/Arithmetic.html">-</a></code>” character is illegal in names, so we consistently sanitize barcodes to contain “<code>_</code>” instead. See <code><a href="https://rdrr.io/r/base/make.names.html">help("make.names")</a></code> for more information on syntactically valid names in <a href="https://www.r-project.org/">R</a>.</li>
<li>
<code>readCellRanger</code> no longer requires reference data defined by <code>refdataDir</code>, although this is still recommended.</li>
</ul>
</div>
<div id="minor-changes-39" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-39" class="anchor"></a>Minor changes</h3>
<ul>
<li>Resaved example datasets.</li>
<li>Switched the example <code>cellranger_small</code> and <code>seurat_small</code> datasets to the publicly available pbmc4k dataset from <a href="https://www.10xgenomics.com/">10X Genomics</a>. Here we’ve subset the top 500 cells and genes by abundance. We’ll use either the pbmc4k or pbmc8k dataset for the vignette in a future update.</li>
<li>
<code>bcbioSingleCell</code> and <code>readCellRanger</code> functions now consistently default to not requiring <code>sampleMetadataFile</code>, which is now <code>NULL</code> by default. For <code>bcbioSingleCell</code>, if a custom sample metadata file is not provided, the function reads from the bcbio YAML metadata. For <code>readCellRanger</code>, the function uses <code>minimalSampleData</code> internally to return minimal metadata, containing <code>sampleName</code> and <code>description</code> columns.</li>
<li>Broke out internal <code>.sampleDirs</code> function to <code>bcbioSingleCell</code> and <code>readCellRanger</code> functions.</li>
<li>Improved <code>plotMarker</code> documentation examples to use mitochondrial genes.</li>
<li>Using <code>seurat_small</code> in place of <code><a href="https://rdrr.io/pkg/Seurat/man/pbmc_small.html">Seurat::pbmc_small</a></code> in working examples.</li>
</ul>
</div>
<div id="internal-code-changes" class="section level3">
<h3 class="hasAnchor">
<a href="#internal-code-changes" class="anchor"></a>Internal code changes</h3>
<ul>
<li>Now consistently using <a href="https://dplyr.tidyverse.org/">dplyr</a> for piped <code>data.frame</code> as much as possible, where applicable. Code is being updated to use <a href="https://dplyr.tidyverse.org/articles/programming.html">tidyeval</a>.</li>
</ul>
</div>
</div>
    <div id="bcbiosinglecell-0115-2018-06-13" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiosinglecell-0115-2018-06-13" class="anchor"></a>bcbioSingleCell 0.1.15 (2018-06-13)</h2>
<div id="minor-changes-38" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-38" class="anchor"></a>Minor changes</h3>
<ul>
<li>Updated internal code to use <code>text</code> as primary argument in <code>markdownHeader</code> calls.</li>
<li>Updated example datasets and unit tests to match.</li>
<li>Working on using pbmc4k as example dataset for vignette.</li>
</ul>
</div>
</div>
    <div id="bcbiosinglecell-0116-2018-06-20" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiosinglecell-0116-2018-06-20" class="anchor"></a>bcbioSingleCell 0.1.16 (2018-06-20)</h2>
<div id="new-functions" class="section level3">
<h3 class="hasAnchor">
<a href="#new-functions" class="anchor"></a>New functions</h3>
<ul>
<li>
<code>cellCountsPerCluster</code> and <code>clusterCellCountsPerSample</code>.</li>
</ul>
</div>
<div id="r-markdown-templates" class="section level3">
<h3 class="hasAnchor">
<a href="#r-markdown-templates" class="anchor"></a>R Markdown templates</h3>
<ul>
<li>Now recommending 1000 UMIs by default, and a maximum of 10% mitochondrial transcripts.</li>
<li>
<a href="http://satijalab.org/seurat/">Seurat</a> clustering now calculates multiple resolutions, suggesting 0.4, 0.8, and 1.2 by default. Dimensional reduction plots have been updated to support looping of these multiple resolutions.</li>
</ul>
</div>
<div id="minor-changes-37" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-37" class="anchor"></a>Minor changes</h3>
<ul>
<li>
<code>as(object, "SingleCellExperiment")</code> coercion now slots stashed metadata into <code>SingleCellExperiment</code>, if defined.</li>
<li>Simplified the internal code for <code>sampleData</code>. <code>seurat</code> objects now share the same code as <code>SingleCellExperiment</code>, and return <code>NULL</code> if the sample data is not defined. The <code>metrics</code> function continues to slot empty metadata in <code>sampleID</code>, <code>sampleName</code>, and <code>interestingGroups</code> if not defined.</li>
<li>
<code>SingleCellExperiment</code> is used as default method over <code>seurat</code> where applicable: <code>fetchData</code> family, <code>plotDimensionalReduction</code> family, <code>plotMarker</code> family, <code>plotFeature</code> family, <code>plotGene</code> family, and <code>plotCellTypesPerCluster</code>. The internal code hasn’t changed, it just is defined primarily for <code>SingleCellExperiment</code>.</li>
<li>
<code>dimRed</code> argument has been renamed to <code>reduction</code>, where applicable.</li>
<li>
<code>topBarcodes</code> can now return either a <code>data.frame</code> or <code>list</code>, containing the top barcodes grouped by sample.</li>
</ul>
</div>
</div>
    <div id="bcbiosinglecell-0117-2018-07-21" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiosinglecell-0117-2018-07-21" class="anchor"></a>bcbioSingleCell 0.1.17 (2018-07-21)</h2>
<div id="major-changes-13" class="section level3">
<h3 class="hasAnchor">
<a href="#major-changes-13" class="anchor"></a>Major changes</h3>
<ul>
<li>
<code>readCellRanger</code> now supports HDF5 data (.h5 files) or MatrixMarket Exchange Format explicitly, using the <code>format</code> argument. The function also now supports user-requested import of raw whitelisted cellular barcodes, using the <code>filtered</code> argument. By default, filtered counts from <a href="https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/what-is-cell-ranger">Cell Ranger</a> are imported.</li>
<li>Removed support for <a href="https://github.com/statOmics/zingeR/">zingeR</a> in <code>diffExp</code>. We will consider adding this functionality back in a future update, once the package is available on <a href="https://bioconductor.org/">Bioconductor</a>. For now, we’re recommending usage of <a href="https://bioconductor.org/packages/zinbwave/">zinbwave</a>.</li>
<li>
<code>filterCells</code> now supports <code>nCells</code> argument for hard cutoffs, applied after other QC filtering cutoffs using <code>nUMI</code>.</li>
<li>Internal data has been renamed to snake_case format, for improved consistency. Note that <code>cell_cycle_markers</code> is used in place of <code>cellCycleMarkers</code>, and <code>cell_type_markers</code> in place of <code>cellTypeMarkers</code>. There’s no easy way to link these files as aliases, so this is a necessary breaking change.</li>
</ul>
</div>
<div id="minor-changes-36" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-36" class="anchor"></a>Minor changes</h3>
<ul>
<li>Improved <a href="http://rmarkdown.rstudio.com/">R Markdown</a> template defaults for quality control and clustering.</li>
<li>Now importing functions related to differential expression into the package NAMESPACE, including <a href="https://bioconductor.org/packages/DESeq2/">DESeq2</a>, <a href="https://bioconductor.org/packages/edgeR/">edgeR</a>, and <a href="https://bioconductor.org/packages/zinbwave/">zinbwave</a>.</li>
<li>Parallelization is no longer used to load samples, improving handling in an HPC cluster environment. We will consider using <a href="https://bioconductor.org/packages/BiocParallel/">BiocParallel</a> in a future update.</li>
<li>Now requiring <a href="https://ggplot2.tidyverse.org/">ggplot2</a> v3.0. Switched to using <code>aes</code> instead of <code>aes_string</code> internally, using new <a href="https://dplyr.tidyverse.org/articles/programming.html">tidyeval</a> syntax.</li>
<li>Removed additional single-cell toolkit packages in Suggests, including <a href="https://bioconductor.org/packages/scater/">scater</a> and <a href="https://bioconductor.org/packages/scran/">scran</a>. This helps speed up build checks on <a href="https://travis-ci.org/">Travis CI</a>.</li>
<li>Reexported functions have been moved to the <a href="https://bioinformatics.sph.harvard.edu/bcbioBase/">bcbioBase</a> package.</li>
<li>Improved internal consistency between the <code>bcbioSingleCell</code> and <code>readCellRanger</code> data import functions.</li>
</ul>
</div>
<div id="additional-notes" class="section level3">
<h3 class="hasAnchor">
<a href="#additional-notes" class="anchor"></a>Additional notes</h3>
<ul>
<li>Package still works with <a href="https://bioconductor.org/">Bioconductor</a> 3.6 / <a href="https://www.r-project.org/">R</a> 3.4, but we will be migrating to <a href="https://bioconductor.org/">Bioconductor</a> 3.7 / <a href="https://www.r-project.org/">R</a> 3.5 when <a href="https://conda.io/">conda</a> r-base is updated to 3.5.1.</li>
</ul>
</div>
</div>
    <div id="bcbiosinglecell-0118-2018-07-31" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiosinglecell-0118-2018-07-31" class="anchor"></a>bcbioSingleCell 0.1.18 (2018-07-31)</h2>
<div id="major-changes-12" class="section level3">
<h3 class="hasAnchor">
<a href="#major-changes-12" class="anchor"></a>Major changes</h3>
<ul>
<li>Now recommending <a href="https://bioconductor.org/packages/BiocManager/">BiocManager</a> instead of <a href="https://bioconductor.org/packages/BiocInstaller/">BiocInstaller</a> to install.</li>
<li>
<code>filterCells</code> now supports <code>zinbwave = TRUE</code> for automatic zero-inflation weights calculation, using the <a href="https://bioconductor.org/packages/zinbwave/">zinbwave</a> package.</li>
<li>Now requiring in <code>bcbioSingleCell</code> object validity checks that all <code>sampleData</code> columns are defined in <code>colData</code>, and that the factor levels also match. If you run into an error because of this change, post an issue on <a href="https://github.com/">GitHub</a> and we’ll sort it out. The validity check method now also checks to make sure that all <code>metricsCols</code> are defined in <code>colData</code>.</li>
<li>
<code>aggregateReplicates</code> now keeps track of interesting groups metadata better and returns a <code>SingleCellExperiment</code> containing this information.</li>
<li>
<code>seurat</code> to <code>SingleCellExperiment</code> now keeps track of <code>rowRanges</code>, if they have been stashed inside the <code>seurat</code> object.</li>
<li>Minimal <code>bcbioSingleCell</code> (<code>indrops_small</code>) and <code>SingleCellExperiment</code> (<code>cellranger_small</code>) objects now contain dimensionality reduction information calculated with <a href="http://satijalab.org/seurat/">Seurat</a>. See the scripts in <code>data-raw/</code> for how this was performed.</li>
<li>
<code>metrics</code>: <code>SingleCellExperiment</code> method now just returns <code>colData</code> with <code>interestingGroups</code> column defined internally with <code>uniteInterestingGroups</code>. Previously we updated the <code>sampleData</code> to long format here dynamically, but this is no longer needed now that <code>colData</code> always must contain sample data in long format.</li>
<li>
<code>sampleData&lt;-</code> assignment method now dynamically updates the sample metadata in <code>colData</code> to match.</li>
<li>Simplified <code>SingleCellExperiment</code> method extension for <code>seurat</code> objects.</li>
<li>Added support for global <a href="https://ggplot2.tidyverse.org/">ggplot2</a> discrete color palettes using <code>bcbio.discrete.color</code> and <code>bcbio.discrete.fill</code> with <code>options</code>.</li>
</ul>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>

</div>


      <footer>
      <div class="copyright">
  <p>Developed by <a href='https://steinbaugh.com/'>Michael Steinbaugh</a>, <a href='https://github.com/roryk/'>Rory Kirchner</a>, <a href='https://github.com/marypiper/'>Mary Piper</a>, <a href='https://github.com/vbarrera/'>Victor Barrera</a>, <a href='https://github.com/sjhosui/'>Shannan Ho Sui</a>, <a href='https://bioinformatics.sph.harvard.edu/'>Harvard Chan Bioinformatics Core</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


