<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Version 0.4 • bcbioSingleCell</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



  <link href="../extra.css" rel="stylesheet">
  

<meta property="og:title" content="Version 0.4" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-news">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">bcbioSingleCell</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.4.14</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/hbc/bcbioSingleCell/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
      <h1 data-toc-skip>Changelog <small>0.4</small></h1>
      
    </div>

    <div id="bcbiosinglecell-040-2019-08-12" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiosinglecell-040-2019-08-12" class="anchor"></a>bcbioSingleCell 0.4.0 (2019-08-12)</h2>
<p>This update is intended to simplify the package and reduce the amount of plotting code that is defined to <code>bcbioSingleCell</code> class specifically, instead of <code>SingleCellExperiment</code>.</p>
<div id="major-changes-1" class="section level3">
<h3 class="hasAnchor">
<a href="#major-changes-1" class="anchor"></a>Major changes</h3>
<ul>
<li>Offloaded plotting functions to AcidPlots, so we can use some of this code in a shared manner with objects imported from 10X Genomics Chromium.</li>
</ul>
</div>
<div id="minor-changes-14" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-14" class="anchor"></a>Minor changes</h3>
<ul>
<li>Improved documentation consistency by using shared roxygen from AcidRoxygen package.</li>
<li>Updated basejump and bcbio R package dependencies.</li>
</ul>
</div>
</div>
    <div id="bcbiosinglecell-041-2019-08-20" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiosinglecell-041-2019-08-20" class="anchor"></a>bcbioSingleCell 0.4.1 (2019-08-20)</h2>
<div id="minor-changes-13" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-13" class="anchor"></a>Minor changes</h3>
<ul>
<li>
<code>plotReadsPerCell</code>: Reworked internal code to use <code>DataFrame</code> primarily instead of <code>tbl_df</code>, so we can remove dplyr dependency.</li>
<li>Updated basejump and bcbioBase dependency versions.</li>
<li>Removed some additional unused dependencies, to lighten the package.</li>
</ul>
</div>
</div>
    <div id="bcbiosinglecell-042-2019-08-22" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiosinglecell-042-2019-08-22" class="anchor"></a>bcbioSingleCell 0.4.2 (2019-08-22)</h2>
<div id="major-changes" class="section level3">
<h3 class="hasAnchor">
<a href="#major-changes" class="anchor"></a>Major changes</h3>
<ul>
<li>
<code>bcbioSingleCell</code>: Simplifed internal <code>colData</code> handling, calling <code>calculateMetrics</code> after creating <code>SingleCellExperiment</code> with <code>makeSingleCellExperiment</code>. This improves consistency with Chromium package and enables better preparation of <code>rowRanges</code> containing spike-ins and transgenes.</li>
</ul>
</div>
<div id="minor-changes-12" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-12" class="anchor"></a>Minor changes</h3>
<ul>
<li>Moved <code>metricsCols</code> global to basejump, so we can share with Chromium.</li>
<li>Improved generator consistency with other basejump and bcbio R packages.</li>
</ul>
</div>
</div>
    <div id="bcbiosinglecell-043-2019-08-27" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiosinglecell-043-2019-08-27" class="anchor"></a>bcbioSingleCell 0.4.3 (2019-08-27)</h2>
<div id="minor-changes-11" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-11" class="anchor"></a>Minor changes</h3>
<ul>
<li>NAMESPACE and basejump dependency updates.</li>
</ul>
</div>
</div>
    <div id="bcbiosinglecell-044-2019-09-09" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiosinglecell-044-2019-09-09" class="anchor"></a>bcbioSingleCell 0.4.4 (2019-09-09)</h2>
<div id="minor-changes-10" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-10" class="anchor"></a>Minor changes</h3>
<ul>
<li>
<code>bcbioSingleCell</code>: Internal generator now calls <code>importSampleData</code> using <code>pipeline = "bcbio"</code> argument, so we don’t run into breaking changes when handling user metadata in a future basejump update.</li>
<li>Updated basejump dependency versions.</li>
</ul>
</div>
</div>
    <div id="bcbiosinglecell-045-2019-09-17" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiosinglecell-045-2019-09-17" class="anchor"></a>bcbioSingleCell 0.4.5 (2019-09-17)</h2>
<div id="minor-changes-9" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-9" class="anchor"></a>Minor changes</h3>
<ul>
<li>Switched QC template to use <code>bcb_file</code> as main file input, similar to bcbioRNASeq QC template.</li>
<li>Updated lintr config.</li>
</ul>
</div>
</div>
    <div id="bcbiosinglecell-046-2019-10-30" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiosinglecell-046-2019-10-30" class="anchor"></a>bcbioSingleCell 0.4.6 (2019-10-30)</h2>
<div id="minor-changes-8" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-8" class="anchor"></a>Minor changes</h3>
<ul>
<li>Updated S4 validity checks and unit tests to support Bioconductor 3.10. Internally, we have to adjust checks against <code>DataFrame</code> class to also look for <code>DFrame</code> class, which has changed in latest S4Vectors update. See related update in bcbioRNASeq package.</li>
<li>Full backward compatiblity with Bioconductor 3.9 is currently maintained.</li>
<li>Updated basejump dependency versions.</li>
</ul>
</div>
</div>
    <div id="bcbiosinglecell-047-2020-01-20" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiosinglecell-047-2020-01-20" class="anchor"></a>bcbioSingleCell 0.4.7 (2020-01-20)</h2>
<div id="minor-changes-7" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-7" class="anchor"></a>Minor changes</h3>
<ul>
<li>Updated messages to use cli package approach. The same approach is now used in the bcbioRNASeq package as well.</li>
<li>Updated basejump dependencies to support rename from bioverbs to AcidGenerics.</li>
</ul>
</div>
</div>
    <div id="bcbiosinglecell-048-2020-02-19" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiosinglecell-048-2020-02-19" class="anchor"></a>bcbioSingleCell 0.4.8 (2020-02-19)</h2>
<div id="minor-changes-6" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-6" class="anchor"></a>Minor changes</h3>
<ul>
<li>Changed license from MIT to GPL-3, matching other bcbio and Acid Genomics R packages.</li>
</ul>
</div>
</div>
    <div id="bcbiosinglecell-049-2020-02-20" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiosinglecell-049-2020-02-20" class="anchor"></a>bcbioSingleCell 0.4.9 (2020-02-20)</h2>
<div id="minor-changes-5" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-5" class="anchor"></a>Minor changes</h3>
<ul>
<li>Tightened up dependencies in DESCRIPTION in preparation for bioconda release update, which breaks with basejump 0.12.0 conda dependency otherwise.</li>
</ul>
</div>
</div>
    <div id="bcbiosinglecell-0410-2020-02-24" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiosinglecell-0410-2020-02-24" class="anchor"></a>bcbioSingleCell 0.4.10 (2020-02-24)</h2>
<div id="minor-changes-4" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-4" class="anchor"></a>Minor changes</h3>
<ul>
<li>
<code>bcbioSingleCell</code>: Removed now defunct <code>spikeNames</code> argument. Refer to <code>makeSummarizedExperiment</code>, <code>makeSingleCellExperiment</code> documentation and release notes in basejump package for details.</li>
</ul>
</div>
</div>
    <div id="bcbiosinglecell-0411-2020-05-11" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiosinglecell-0411-2020-05-11" class="anchor"></a>bcbioSingleCell 0.4.11 (2020-05-11)</h2>
<div id="minor-changes-3" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-3" class="anchor"></a>Minor changes</h3>
<ul>
<li>
<code>updateObject</code>: Bug fix release for breaking changes introduced by <code>colData&lt;-</code> assignment on <code>SingleCellExperiment</code> / <code>SummarizedExperiment</code> expecting <code>check = FALSE</code> support for downstream <code>updateObject</code> call. Added <code>...</code> passthrough that prevents this error in method support for <code>bcbioSingleCell</code> class. This should now be fully compliant with R 4.0 / Bioconductor 3.11 release.</li>
<li>
<code>updateObject</code>: Added <code>verbose</code> argument support, defaulting now to <code>FALSE</code>.</li>
</ul>
</div>
</div>
    <div id="bcbiosinglecell-0412-2020-05-12" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiosinglecell-0412-2020-05-12" class="anchor"></a>bcbioSingleCell 0.4.12 (2020-05-12)</h2>
<div id="minor-changes-2" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-2" class="anchor"></a>Minor changes</h3>
<ul>
<li>Bug fix for integer handling in <code>.importReads</code> following switch to <code><a href="https://vroom.r-lib.org/reference/vroom.html">vroom::vroom</a></code> from <code><a href="https://Rdatatable.gitlab.io/data.table/reference/fread.html">data.table::fread</a></code>.</li>
</ul>
</div>
</div>
    <div id="bcbiosinglecell-0413-2020-07-24" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiosinglecell-0413-2020-07-24" class="anchor"></a>bcbioSingleCell 0.4.13 (2020-07-24)</h2>
<div id="minor-changes-1" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-1" class="anchor"></a>Minor changes</h3>
<ul>
<li>Maintenance release, updating minimum R dependency to 4.0.</li>
</ul>
</div>
</div>
    <div id="bcbiosinglecell-0414-2020-10-08" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiosinglecell-0414-2020-10-08" class="anchor"></a>bcbioSingleCell 0.4.14 (2020-10-08)</h2>
<div id="minor-changes" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes" class="anchor"></a>Minor changes</h3>
<ul>
<li>Maintenance release, updating dependency version requirements.</li>
</ul>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>

</div>


      <footer>
      <div class="copyright">
  <p>Developed by <a href='https://steinbaugh.com/'>Michael Steinbaugh</a>, <a href='https://github.com/roryk/'>Rory Kirchner</a>, <a href='https://github.com/marypiper/'>Mary Piper</a>, <a href='https://github.com/vbarrera/'>Victor Barrera</a>, <a href='https://github.com/sjhosui/'>Shannan Ho Sui</a>, <a href='https://bioinformatics.sph.harvard.edu/'>Harvard Chan Bioinformatics Core</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


