<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Version 0.3 • bcbioSingleCell</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



  <link href="../extra.css" rel="stylesheet">
  

<meta property="og:title" content="Version 0.3" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-news">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">bcbioSingleCell</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.4.14</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/hbc/bcbioSingleCell/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
      <h1 data-toc-skip>Changelog <small>0.3</small></h1>
      
    </div>

    <div id="bcbiosinglecell-030-2018-11-06" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiosinglecell-030-2018-11-06" class="anchor"></a>bcbioSingleCell 0.3.0 (2018-11-06)</h2>
<p>Fork of <code>hbc/bcbioSingleCell</code> code base, for additional development and improvements prior to Bioconductor submission. The v0.2 release series is being maintained on the <code>hbc</code> organization page for stability.</p>
<p>The v0.3 release series serves to offload some of the code base to the <a href="https://basejump.acidgenomics.com/">basejump</a> package, freeing <a href="https://bioinformatics.sph.harvard.edu/bcbioSingleCell/">bcbioSingleCell</a> up to be lighterweight and easier to maintain long-term.</p>
<p>Note that v0.2 release series will remain pinned to basejump v0.7 series.</p>
<p>CellRanger code will be split out into a separate Chromium R package in a future update, so some of these <code>SingleCellExperiment</code>-specific methods need to be consolidated into the basejump package.</p>
<div id="major-changes-9" class="section level3">
<h3 class="hasAnchor">
<a href="#major-changes-9" class="anchor"></a>Major changes</h3>
<ul>
<li>
<code>aggregateReplicates</code>, <code>cell2sample</code>, <code>combine</code>, <code>plotZerosVsDepth</code>, <code>sampleData</code>, <code>selectSamples</code>, <code>subsetPerSample</code>, <code>topBarcodes</code>: Offloaded to basejump as <code>SingleCellExperiment</code> S4 methods.</li>
<li>Removed generics offloaded to basejump.</li>
<li>
<code>barcodeRanksPerSample</code>: Reworked internal code, using improved <code>countsPerSample</code> and <code>ranks</code> calculations.</li>
<li>Updated extract method to work on <code>SingleCellExperiment</code>, so it can also be inherited for <code>CellRanger</code> S4 class.</li>
<li>Reorganized <code>metricsPerSample</code> code approach.</li>
</ul>
</div>
<div id="minor-changes-32" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-32" class="anchor"></a>Minor changes</h3>
<ul>
<li>Consolidated S4 validity check code into <code>AllClasses.R</code> file.</li>
<li>Updated Travis and AppVeyor CI configuration.</li>
<li>Split out some internal code into <code>barcodes-internal.R</code>. This includes <code>.nCount</code> and <code>.rawMetrics</code>.</li>
<li>Improved documentation for <code>bcbioSingleCell</code> generator function.</li>
<li>Reorganized internal CellRanger processing functions.</li>
<li>Improved documentation for <code>CellRanger</code> generator function.</li>
<li>Reworked plotting function code to integrate with basejump package.</li>
<li>Split out plotting code methods into internal functions (e.g. <code>plotMitoVsCoding.SingleCellExperiment</code>).</li>
</ul>
</div>
</div>
    <div id="bcbiosinglecell-031-2018-11-14" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiosinglecell-031-2018-11-14" class="anchor"></a>bcbioSingleCell 0.3.1 (2018-11-14)</h2>
<div id="major-changes-8" class="section level3">
<h3 class="hasAnchor">
<a href="#major-changes-8" class="anchor"></a>Major changes</h3>
<ul>
<li>Updated NAMESPACE to inherit from basejump instead of any of the basejump subpackages, which are subject to rework during this release series.</li>
<li>Renamed example data from <code>cellranger_small</code> to <code>cellranger</code>, and <code>indrops_small</code> to simply <code>indrops</code>.</li>
</ul>
</div>
<div id="minor-changes-31" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-31" class="anchor"></a>Minor changes</h3>
<ul>
<li>Transitioning to goalie for assert checks in place of assertive package.</li>
<li>Miscellaneous documentation improvements.</li>
<li>Updated working examples for plotting functions, using <code>indrops</code> example data.</li>
<li>Reworked <code>bcbioSingleCell</code> <code>show</code> (<code>print</code>) method.</li>
</ul>
</div>
</div>
    <div id="bcbiosinglecell-032-2018-11-19" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiosinglecell-032-2018-11-19" class="anchor"></a>bcbioSingleCell 0.3.2 (2018-11-19)</h2>
<div id="major-changes-7" class="section level3">
<h3 class="hasAnchor">
<a href="#major-changes-7" class="anchor"></a>Major changes</h3>
<ul>
<li>Migrating <code>CellRanger</code> S4 class to separate Chromium R package.</li>
<li>Migrated additional S4 generics to basejump package (bioverbs).</li>
<li>Exporting <code>calculateMetrics</code> as a user-accessible function.</li>
<li>Extract (<code><a href="https://rdrr.io/r/base/Extract.html">[</a></code>) now returns <code>bcbioSingleCell</code> object again.</li>
<li>
<code>filterCells</code>: Reworking to return <code>SingleCellExperiment</code> instead of <code>bcbioSingleCell</code> object.</li>
<li>Improved plotting functions to inherit params from basejump.</li>
</ul>
</div>
<div id="minor-changes-30" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-30" class="anchor"></a>Minor changes</h3>
<ul>
<li>Removed <code>barcodePattern</code> global variable.</li>
</ul>
</div>
</div>
    <div id="bcbiosinglecell-033-2018-11-25" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiosinglecell-033-2018-11-25" class="anchor"></a>bcbioSingleCell 0.3.3 (2018-11-25)</h2>
<div id="minor-changes-29" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-29" class="anchor"></a>Minor changes</h3>
<ul>
<li>Updated basejump and bcbioBase dependencies, to tighten up build checks.</li>
<li>Documentation and CI configuration fixes.</li>
</ul>
</div>
</div>
    <div id="bcbiosinglecell-034-2018-12-01" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiosinglecell-034-2018-12-01" class="anchor"></a>bcbioSingleCell 0.3.4 (2018-12-01)</h2>
<div id="major-changes-6" class="section level3">
<h3 class="hasAnchor">
<a href="#major-changes-6" class="anchor"></a>Major changes</h3>
<ul>
<li>
<code>metrics</code>: Removed code and consolidated <code>SingleCellExperiment</code> method support in basejump package.</li>
</ul>
</div>
<div id="minor-changes-28" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-28" class="anchor"></a>Minor changes</h3>
<ul>
<li>Documentation improvements.</li>
<li>
<code>plotQC</code>: Improved validity checks.</li>
</ul>
</div>
</div>
    <div id="bcbiosinglecell-035-2018-12-22" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiosinglecell-035-2018-12-22" class="anchor"></a>bcbioSingleCell 0.3.5 (2018-12-22)</h2>
<p>This release reworks the internal assert checks substantially.</p>
<div id="major-changes-5" class="section level3">
<h3 class="hasAnchor">
<a href="#major-changes-5" class="anchor"></a>Major changes</h3>
<ul>
<li>First release that has fully migrated to using goalie package instead of assertive package internally for checks.</li>
<li>Reworked <code>bcbioSingleCell</code> S4 validity checks.</li>
<li>Reworked <code>bcbioSingleCell</code> generator function to use cleaner assert checks.</li>
<li>Migrated to inheriting S4 generics from bioverbs, where applicable.</li>
</ul>
</div>
<div id="minor-changes-27" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-27" class="anchor"></a>Minor changes</h3>
<ul>
<li>Reworked documentation to use sentence case instead of title case.</li>
</ul>
</div>
</div>
    <div id="bcbiosinglecell-036-2019-01-13" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiosinglecell-036-2019-01-13" class="anchor"></a>bcbioSingleCell 0.3.6 (2019-01-13)</h2>
<div id="minor-changes-26" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-26" class="anchor"></a>Minor changes</h3>
<ul>
<li>Updated basejump and bcbioBase dependencies, to pass build checks.</li>
<li>Split out imports into a separate R file.</li>
</ul>
</div>
</div>
    <div id="bcbiosinglecell-037-2019-02-12" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiosinglecell-037-2019-02-12" class="anchor"></a>bcbioSingleCell 0.3.7 (2019-02-12)</h2>
<div id="minor-changes-25" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-25" class="anchor"></a>Minor changes</h3>
<ul>
<li>
<code>bcbioSingleCell</code> generator: improved internal handling of <code>mapCellsToSamples</code> call. This reworked join step checks the sampleID mapping more carefully.</li>
<li>Removed <code>plotGene</code> from deprecations. This function has been renamed to <code>plotCounts</code> instead in the basejump package.</li>
</ul>
</div>
</div>
    <div id="bcbiosinglecell-038-2019-03-18" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiosinglecell-038-2019-03-18" class="anchor"></a>bcbioSingleCell 0.3.8 (2019-03-18)</h2>
<div id="minor-changes-24" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-24" class="anchor"></a>Minor changes</h3>
<ul>
<li>Updated basejump and bcbioBase dependencies.</li>
<li>Updated documentation.</li>
<li>Added complete release information in NEWS.</li>
</ul>
</div>
</div>
    <div id="bcbiosinglecell-039-2019-04-01" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiosinglecell-039-2019-04-01" class="anchor"></a>bcbioSingleCell 0.3.9 (2019-04-01)</h2>
<div id="minor-changes-23" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-23" class="anchor"></a>Minor changes</h3>
<ul>
<li>Migrated development code to <a href="https://acidgenomics.com/">Acid Genomics</a>.</li>
<li>Updated <a href="https://basejump.acidgenomics.com/">basejump</a> dependency to v0.10 release series.</li>
</ul>
</div>
</div>
    <div id="bcbiosinglecell-0310-2019-04-07" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiosinglecell-0310-2019-04-07" class="anchor"></a>bcbioSingleCell 0.3.10 (2019-04-07)</h2>
<div id="major-changes-4" class="section level3">
<h3 class="hasAnchor">
<a href="#major-changes-4" class="anchor"></a>Major changes</h3>
<ul>
<li>
<code>updateObject</code> is now fully backward compatible for all objects created by the package, including versions prior to v0.1.</li>
<li>Tightened up S4 validity class checks for <code>bcbioSingleCell</code>. Similar to <code>bcbioRNASeq</code> class checks, now requiring these elements in <code>metadata</code>: <code>bcbioCommandsLog</code>, <code>bcbioLog</code>, <code>dataVersions</code>, <code>gffFile</code>, <code>lanes</code>, <code>programVersions</code>, <code>projectDir</code>, <code>runDate</code>, <code>wd</code>, and <code>yaml</code>. Previously, these weren’t required because we were sharing internal validity code checks with the <code>CellRanger</code> S4 class. <code>CellRanger</code> has since been moved to the new Chromium package, so it’s now appropriate to tighten up bcbioSingleCell.</li>
<li>Extract method <code><a href="https://rdrr.io/r/base/Extract.html">[</a></code> on <code>bcbioSingleCell</code> class objects no longer alters the version of the object. It only sets <code>subset = TRUE</code>, similar to bcbioRNASeq.</li>
</ul>
</div>
<div id="minor-changes-22" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-22" class="anchor"></a>Minor changes</h3>
<ul>
<li>Bug fix release for formal rename in <code>emptyRanges</code> from <code>mcolsNames</code> to <code>mcolnames</code>. This has been changed in <a href="https://basejump.acidgenomics.com/">basejump</a> 0.10.3.</li>
</ul>
</div>
</div>
    <div id="bcbiosinglecell-0311-2019-04-11" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiosinglecell-0311-2019-04-11" class="anchor"></a>bcbioSingleCell 0.3.11 (2019-04-11)</h2>
<div id="major-changes-3" class="section level3">
<h3 class="hasAnchor">
<a href="#major-changes-3" class="anchor"></a>Major changes</h3>
<ul>
<li>Restricted S4 methods back to <code>bcbioSingleCell</code> from <code>SingleCellExperiment</code> where applicable. Generally, this applies to the supported QC plotting functions. Previously, the methods were defined against <code>SingleCellExperiment</code> to also support <code>CellRanger</code> S4 class, which has since been moved to the <a href="https://chromium.acidgenomics.com/">Chromium</a> package.</li>
</ul>
</div>
<div id="minor-changes-21" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-21" class="anchor"></a>Minor changes</h3>
<ul>
<li>Improved code coverage by adding additional unit tests.</li>
<li>Updated internal reexport method for S4 generics, making <code>reexports.Rd</code> file no longer necessary.</li>
<li>Renamed R Markdown QC template from <code>quality_control</code> to <code>quality-control</code>, using kebab case instead of snake case.</li>
</ul>
</div>
</div>
    <div id="bcbiosinglecell-0312-2019-04-18" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiosinglecell-0312-2019-04-18" class="anchor"></a>bcbioSingleCell 0.3.12 (2019-04-18)</h2>
<div id="minor-changes-20" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-20" class="anchor"></a>Minor changes</h3>
<ul>
<li>Deprecated <code>prepareSingleCellTemplate</code>, in favor of <code>prepareTemplate</code>.</li>
<li>Switched Travis CI configuration to use <code>singlecell</code> Docker image.</li>
</ul>
</div>
</div>
    <div id="bcbiosinglecell-0313-2019-04-23" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiosinglecell-0313-2019-04-23" class="anchor"></a>bcbioSingleCell 0.3.13 (2019-04-23)</h2>
<div id="minor-changes-19" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-19" class="anchor"></a>Minor changes</h3>
<ul>
<li>Now importing ggplot2 code from <a href="https://AcidPlots.acidgenomics.com/">AcidPlots</a> package.</li>
<li>
<code>plotCellCounts</code>: Improved default appearance, removing black border.</li>
<li>Updated basejump and bcbio R package dependencies.</li>
</ul>
</div>
</div>
    <div id="bcbiosinglecell-0314-2019-04-25" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiosinglecell-0314-2019-04-25" class="anchor"></a>bcbioSingleCell 0.3.14 (2019-04-25)</h2>
<ul>
<li>S4 generic reexport documentation fixes.</li>
</ul>
</div>
    <div id="bcbiosinglecell-0315-2019-05-29" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiosinglecell-0315-2019-05-29" class="anchor"></a>bcbioSingleCell 0.3.15 (2019-05-29)</h2>
<div id="minor-changes-18" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-18" class="anchor"></a>Minor changes</h3>
<ul>
<li>Updated <code>barcodeRanksPerSample</code> and <code>plotBarcodeRanks</code> to support DropletUtils update in Bioconductor 3.9.</li>
<li>Improved Travis and AppVeyor CI configuration.</li>
</ul>
</div>
</div>
    <div id="bcbiosinglecell-0316-2019-07-17" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiosinglecell-0316-2019-07-17" class="anchor"></a>bcbioSingleCell 0.3.16 (2019-07-17)</h2>
<div id="minor-changes-17" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-17" class="anchor"></a>Minor changes</h3>
<ul>
<li>Updated basejump dependencies.</li>
<li>Improved Travis CI docker configuration.</li>
</ul>
</div>
</div>
    <div id="bcbiosinglecell-0317-2019-07-24" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiosinglecell-0317-2019-07-24" class="anchor"></a>bcbioSingleCell 0.3.17 (2019-07-24)</h2>
<div id="minor-changes-16" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-16" class="anchor"></a>Minor changes</h3>
<ul>
<li>Improved internal S4 function naming consistency.</li>
<li>Updated basejump dependencies.</li>
</ul>
</div>
</div>
    <div id="bcbiosinglecell-0318-2019-07-29" class="section level2">
<h2 class="hasAnchor">
<a href="#bcbiosinglecell-0318-2019-07-29" class="anchor"></a>bcbioSingleCell 0.3.18 (2019-07-29)</h2>
<div id="major-changes-2" class="section level3">
<h3 class="hasAnchor">
<a href="#major-changes-2" class="anchor"></a>Major changes</h3>
<ul>
<li>Updated R dependency to 3.6. This now requires Bioconductor 3.9+.</li>
<li>
<code>updateObject</code>: Simplified method support, dropping <code>rowRanges</code> argument. No longer attempting to coerce assays back to a simple list, which doesn’t work well on Bioconductor 3.10 Devel currently.</li>
</ul>
</div>
<div id="minor-changes-15" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-15" class="anchor"></a>Minor changes</h3>
<ul>
<li>Updated basejump dependencies.</li>
<li>Improved code coverage over 90%.</li>
</ul>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>

</div>


      <footer>
      <div class="copyright">
  <p>Developed by <a href='https://steinbaugh.com/'>Michael Steinbaugh</a>, <a href='https://github.com/roryk/'>Rory Kirchner</a>, <a href='https://github.com/marypiper/'>Mary Piper</a>, <a href='https://github.com/vbarrera/'>Victor Barrera</a>, <a href='https://github.com/sjhosui/'>Shannan Ho Sui</a>, <a href='https://bioinformatics.sph.harvard.edu/'>Harvard Chan Bioinformatics Core</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


