<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>bcbio single-cell RNA-seq data set — bcbioSingleCell • bcbioSingleCell</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



  <link href="../extra.css" rel="stylesheet">
  

<meta property="og:title" content="bcbio single-cell RNA-seq data set — bcbioSingleCell" />
<meta property="og:description" content="bcbioSingleCell is an S4 class that extends SingleCellExperiment, and is
designed to store a bcbio single-cell RNA-seq analysis. This class contains
read counts saved as a sparse matrix (sparseMatrix), sample metadata, and
cell quality control metrics." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">bcbioSingleCell</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.4.13</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/hbc/bcbioSingleCell/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>bcbio single-cell RNA-seq data set</h1>
    <small class="dont-index">Source: <a href='https://github.com/hbc/bcbioSingleCell/blob/master/R/AllGenerators.R'><code>R/AllGenerators.R</code></a></small>
    <div class="hidden name"><code>bcbioSingleCell.Rd</code></div>
    </div>

    <div class="ref-description">
    <p><code>bcbioSingleCell</code> is an S4 class that extends <code>SingleCellExperiment</code>, and is
designed to store a bcbio single-cell RNA-seq analysis. This class contains
read counts saved as a sparse matrix (<code>sparseMatrix</code>), sample metadata, and
cell quality control metrics.</p>
    </div>

    <pre class="usage"><span class='fu'>bcbioSingleCell</span><span class='op'>(</span>
  <span class='va'>uploadDir</span>,
  sampleMetadataFile <span class='op'>=</span> <span class='cn'>NULL</span>,
  organism <span class='op'>=</span> <span class='cn'>NULL</span>,
  ensemblRelease <span class='op'>=</span> <span class='cn'>NULL</span>,
  genomeBuild <span class='op'>=</span> <span class='cn'>NULL</span>,
  gffFile <span class='op'>=</span> <span class='cn'>NULL</span>,
  transgeneNames <span class='op'>=</span> <span class='cn'>NULL</span>,
  interestingGroups <span class='op'>=</span> <span class='st'>"sampleName"</span>,
  BPPARAM <span class='op'>=</span> <span class='fu'>BiocParallel</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/BiocParallel/man/SerialParam-class.html'>SerialParam</a></span><span class='op'>(</span><span class='op'>)</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>uploadDir</th>
      <td><p><code><a href='https://rdrr.io/r/base/character.html'>character(1)</a></code>.
Final upload directory path.</p></td>
    </tr>
    <tr>
      <th>sampleMetadataFile</th>
      <td><p><code><a href='https://rdrr.io/r/base/character.html'>character(1)</a></code>.
Sample metadata file path. CSV or TSV is preferred, but Excel worksheets
are also supported. Check the documentation for conventions and required
columns.</p></td>
    </tr>
    <tr>
      <th>organism</th>
      <td><p><code><a href='https://rdrr.io/r/base/character.html'>character(1)</a></code>.
Full Latin organism name (e.g. <code>"Homo sapiens"</code>).</p></td>
    </tr>
    <tr>
      <th>ensemblRelease</th>
      <td><p><code><a href='https://rdrr.io/r/base/integer.html'>integer(1)</a></code>.
Ensembl release version (e.g. <code>90</code>). We recommend setting this value if
possible, for improved reproducibility. When left unset, the latest release
available via AnnotationHub/ensembldb is used. Note that the latest version
available can vary, depending on the versions of AnnotationHub and
ensembldb in use.</p></td>
    </tr>
    <tr>
      <th>genomeBuild</th>
      <td><p><code><a href='https://rdrr.io/r/base/character.html'>character(1)</a></code>.
Ensembl genome build assembly name (e.g. <code>"GRCh38"</code>). If set <code>NULL</code>,
defaults to the most recent build available.
Note: don't pass in UCSC build IDs (e.g. <code>"hg38"</code>).</p></td>
    </tr>
    <tr>
      <th>gffFile</th>
      <td><p><code><a href='https://rdrr.io/r/base/character.html'>character(1)</a></code>.
GFF/GTF (General Feature Format) file.
Generally, we recommend using GTF (GFFv2) instead of GFFv3.</p></td>
    </tr>
    <tr>
      <th>transgeneNames</th>
      <td><p><code>character</code>.
Vector indicating which assay rows denote transgenes (e.g. EGFP, TDTOMATO).</p></td>
    </tr>
    <tr>
      <th>interestingGroups</th>
      <td><p><code>character</code>.
Groups of interest to use for visualization.
Corresponds to factors describing the columns of the object.</p></td>
    </tr>
    <tr>
      <th>BPPARAM</th>
      <td><p>An optional <code><a href='https://rdrr.io/pkg/BiocParallel/man/BiocParallelParam-class.html'>BiocParallelParam</a></code> instance
    determining the parallel back-end to be used during evaluation, or a
    <code>list</code> of <code>BiocParallelParam</code> instances, to be applied in
    sequence for nested calls to <span class="pkg">BiocParallel</span> functions.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p><code>bcbioSingleCell</code>.</p>
    <h2 class="hasAnchor" id="note"><a class="anchor" href="#note"></a>Note</h2>

    <p>Updated 2020-02-24.</p>
    <h2 class="hasAnchor" id="remote-data"><a class="anchor" href="#remote-data"></a>Remote data</h2>

    


<p>When working in RStudio, we recommend connecting to the bcbio-nextgen run
directory as a remote connection over
<a href='https://github.com/osxfuse/osxfuse/wiki/SSHFS'>sshfs</a>.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'>
<ul>
<li><p><code><a href='https://rdrr.io/pkg/SingleCellExperiment/man/SingleCellExperiment.html'>SingleCellExperiment::SingleCellExperiment()</a></code>.</p></li>
<li><p><code>.S4methods(class = "bcbioSingleCell")</code>.</p></li>
</ul>
</div>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Michael Steinbaugh</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='va'>uploadDir</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>"extdata/indrops"</span>, package <span class='op'>=</span> <span class='st'>"bcbioSingleCell"</span><span class='op'>)</span>

<span class='va'>x</span> <span class='op'>&lt;-</span> <span class='fu'>bcbioSingleCell</span><span class='op'>(</span><span class='va'>uploadDir</span><span class='op'>)</span>
</div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'><span style='color: #00BBBB;'>──</span><span> </span><span style='font-weight: bold;'>bcbioSingleCell</span><span> </span><span style='color: #00BBBB;'>──────────────────────────────────</span><span></span></div><div class='output co'>#&gt; <span class='message'>Importing bcbio-nextgen single-cell RNA-seq run</span></div><div class='output co'>#&gt; <span class='message'>projectDir: 2018-01-01_bcbio</span></div><div class='output co'>#&gt; <span class='message'>1 sample detected:</span></div><div class='output co'>#&gt; <span class='message'>    ● multiplexed_AAAAAAAA</span></div><div class='output co'>#&gt; <span class='message'>→ Importing </span><span style='color: #0000BB;'>project-summary.yaml</span><span> at </span><span style='color: #0000BB;'>/private/var/folders/l1/8y8sjzmn15v49jgrqglghcfr0000gn/T/Rtmpj5yP1c/temp_libpathe10850cb4eb4/bcbioSingleCell/extdata/indrops/2018-01-01_bcbio</span><span> using </span><span style='color: #0000BB;'>yaml</span><span>::</span><span style='color: #000000;background-color: #BBBBBB;'>`yaml.load_file()`</span><span>.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #BBBB00;'>!</span><span> Data versions are missing.</span></div><div class='output co'>#&gt; <span class='message'>→ Importing </span><span style='color: #0000BB;'>programs.txt</span><span> at </span><span style='color: #0000BB;'>/private/var/folders/l1/8y8sjzmn15v49jgrqglghcfr0000gn/T/Rtmpj5yP1c/temp_libpathe10850cb4eb4/bcbioSingleCell/extdata/indrops/2018-01-01_bcbio</span><span> using </span><span style='color: #0000BB;'>vroom</span><span>::</span><span style='color: #000000;background-color: #BBBBBB;'>`vroom()`</span><span>.</span></div><div class='output co'>#&gt; <span class='message'>→ Importing </span><span style='color: #0000BB;'>bcbio-nextgen.log</span><span> at </span><span style='color: #0000BB;'>/private/var/folders/l1/8y8sjzmn15v49jgrqglghcfr0000gn/T/Rtmpj5yP1c/temp_libpathe10850cb4eb4/bcbioSingleCell/extdata/indrops/2018-01-01_bcbio</span><span> using </span><span style='color: #0000BB;'>readr</span><span>::</span><span style='color: #000000;background-color: #BBBBBB;'>`read_lines()`</span><span>.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #BBBB00;'>!</span><span> </span><span style='color: #0000BB;'>bcbio-nextgen.log</span><span> file is empty.</span></div><div class='output co'>#&gt; <span class='message'>→ Importing </span><span style='color: #0000BB;'>bcbio-nextgen-commands.log</span><span> at </span><span style='color: #0000BB;'>/private/var/folders/l1/8y8sjzmn15v49jgrqglghcfr0000gn/T/Rtmpj5yP1c/temp_libpathe10850cb4eb4/bcbioSingleCell/extdata/indrops/2018-01-01_bcbio</span><span> using </span><span style='color: #0000BB;'>readr</span><span>::</span><span style='color: #000000;background-color: #BBBBBB;'>`read_lines()`</span><span>.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> 1000 reads per cellular barcode cutoff detected.</span></div><div class='output co'>#&gt; <span class='message'>Counts will imported as genes.</span></div><div class='output co'>#&gt; <span class='message'>UMI type: harvard-indrop-v3</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>── </span><span style='font-weight: bold;'>Sample metadata</span><span> ──</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>── </span><span style='font-weight: bold;'>Counts</span><span> ──</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>→ Importing counts.</span></div><div class='output co'>#&gt; <span class='message'>→ Importing </span><span style='color: #0000BB;'>multiplexed-AAAAAAAA.mtx.rownames</span><span> at </span><span style='color: #0000BB;'>/private/var/folders/l1/8y8sjzmn15v49jgrqglghcfr0000gn/T/Rtmpj5yP1c/temp_libpathe10850cb4eb4/bcbioSingleCell/extdata/indrops/multiplexed-AAAAAAAA</span><span> using </span><span style='color: #0000BB;'>readr</span><span>::</span><span style='color: #000000;background-color: #BBBBBB;'>`read_lines()`</span><span>.</span></div><div class='output co'>#&gt; <span class='message'>→ Importing </span><span style='color: #0000BB;'>multiplexed-AAAAAAAA.mtx.colnames</span><span> at </span><span style='color: #0000BB;'>/private/var/folders/l1/8y8sjzmn15v49jgrqglghcfr0000gn/T/Rtmpj5yP1c/temp_libpathe10850cb4eb4/bcbioSingleCell/extdata/indrops/multiplexed-AAAAAAAA</span><span> using </span><span style='color: #0000BB;'>readr</span><span>::</span><span style='color: #000000;background-color: #BBBBBB;'>`read_lines()`</span><span>.</span></div><div class='output co'>#&gt; <span class='message'>→ Importing </span><span style='color: #0000BB;'>multiplexed-AAAAAAAA.mtx</span><span> at </span><span style='color: #0000BB;'>/private/var/folders/l1/8y8sjzmn15v49jgrqglghcfr0000gn/T/Rtmpj5yP1c/temp_libpathe10850cb4eb4/bcbioSingleCell/extdata/indrops/multiplexed-AAAAAAAA</span><span> using </span><span style='color: #0000BB;'>Matrix</span><span>::</span><span style='color: #000000;background-color: #BBBBBB;'>`readMM()`</span><span>.</span></div><div class='output co'>#&gt; <span class='message'>→ Importing sidecar </span><span style='color: #0000BB;'>multiplexed-AAAAAAAA.mtx.rownames</span><span> at </span><span style='color: #0000BB;'>/private/var/folders/l1/8y8sjzmn15v49jgrqglghcfr0000gn/T/Rtmpj5yP1c/temp_libpathe10850cb4eb4/bcbioSingleCell/extdata/indrops/multiplexed-AAAAAAAA</span><span>.</span></div><div class='output co'>#&gt; <span class='message'>→ Importing </span><span style='color: #0000BB;'>multiplexed-AAAAAAAA.mtx.rownames</span><span> at </span><span style='color: #0000BB;'>/private/var/folders/l1/8y8sjzmn15v49jgrqglghcfr0000gn/T/Rtmpj5yP1c/temp_libpathe10850cb4eb4/bcbioSingleCell/extdata/indrops/multiplexed-AAAAAAAA</span><span> using </span><span style='color: #0000BB;'>readr</span><span>::</span><span style='color: #000000;background-color: #BBBBBB;'>`read_lines()`</span><span>.</span></div><div class='output co'>#&gt; <span class='message'>→ Importing sidecar </span><span style='color: #0000BB;'>multiplexed-AAAAAAAA.mtx.colnames</span><span> at </span><span style='color: #0000BB;'>/private/var/folders/l1/8y8sjzmn15v49jgrqglghcfr0000gn/T/Rtmpj5yP1c/temp_libpathe10850cb4eb4/bcbioSingleCell/extdata/indrops/multiplexed-AAAAAAAA</span><span>.</span></div><div class='output co'>#&gt; <span class='message'>→ Importing </span><span style='color: #0000BB;'>multiplexed-AAAAAAAA.mtx.colnames</span><span> at </span><span style='color: #0000BB;'>/private/var/folders/l1/8y8sjzmn15v49jgrqglghcfr0000gn/T/Rtmpj5yP1c/temp_libpathe10850cb4eb4/bcbioSingleCell/extdata/indrops/multiplexed-AAAAAAAA</span><span> using </span><span style='color: #0000BB;'>readr</span><span>::</span><span style='color: #000000;background-color: #BBBBBB;'>`read_lines()`</span><span>.</span></div><div class='output co'>#&gt; <span class='message'>→ Imported </span><span style='color: #0000BB;'>multiplexed-AAAAAAAA</span><span>.</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>── </span><span style='font-weight: bold;'>Feature metadata</span><span> ──</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>bcbio GTF file:</span>
#&gt; <span class='message'>  /n/app/bcbio/dev/genomes/Hsapiens/hg38/rnaseq/ref-transcripts.gtf</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #BBBB00;'>!</span><span> bcbio GTF file is not accessible.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #BBBB00;'>!</span><span> Slotting empty ranges into </span><span style='color: #000000;background-color: #BBBBBB;'>`rowRanges()`</span><span>.</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>── </span><span style='font-weight: bold;'>Column data</span><span> ──</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'></span><span style='color: #BBBB00;'>!</span><span> </span><span style='color: #000000;background-color: #BBBBBB;'>`sampleMetadataFile`</span><span> is recommended for multiplexed samples (e.g. </span><span style='color: #0000BB;'>harvard-indrop-v3</span><span>).</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>── </span><span style='font-weight: bold;'>Metadata</span><span> ──</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>→ Importing unfiltered cellular barcode distributions.</span></div><div class='output co'>#&gt; <span class='message'>→ Importing </span><span style='color: #0000BB;'>multiplexed-AAAAAAAA-barcodes.tsv</span><span> at </span><span style='color: #0000BB;'>/private/var/folders/l1/8y8sjzmn15v49jgrqglghcfr0000gn/T/Rtmpj5yP1c/temp_libpathe10850cb4eb4/bcbioSingleCell/extdata/indrops/multiplexed-AAAAAAAA</span><span> using </span><span style='color: #0000BB;'>vroom</span><span>::</span><span style='color: #000000;background-color: #BBBBBB;'>`vroom()`</span><span>.</span></div><div class='output co'>#&gt; <span class='message'>→ Calculating 100 sample metrics.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #BBBB00;'>!</span><span> Calculating metrics without biotype information.</span>
#&gt; <span class='message'></span><span style='color: #000000;background-color: #BBBBBB;'>`rowRanges()`</span><span> required to calculate: </span><span style='color: #000000;background-color: #BBBBBB;'>`nCoding, nMito, mitoRatio`</span><span>.</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BB00;'>✓</span><span> bcbio single-cell RNA-seq run imported successfully.</span></div><div class='input'><span class='fu'><a href='https://rdrr.io/r/base/print.html'>print</a></span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span>
</div><div class='output co'>#&gt; bcbioSingleCell 0.4.13
#&gt; uploadDir: /private/var/folders/l1/8y8sjzmn15v49jgrqglghcfr0000gn/T/Rtmpj5yP1c/temp_libpathe10850cb4eb4/bcbioSingleCell/extdata/indrops
#&gt; dates(2): [bcbio] 2018-01-01; [R] 2020-10-08
#&gt; level: genes
#&gt; interestingGroups: sampleName
#&gt; filtered: FALSE
#&gt; class: SingleCellExperiment 
#&gt; dim: 50 100 
#&gt; metadata(27): allSamples bcbioCommandsLog ... wd yaml
#&gt; assays(1): counts
#&gt; rownames(50): ENSG00000071082 ENSG00000100316 ... ENSG00000269028
#&gt;   ENSG00000282105
#&gt; rowData names(0):
#&gt; colnames(100): AAACACTA_CTTCGATT AAACTACA_CCACATTA ...
#&gt;   TGGGAATT_ATATAGGA TGTTATCA_ACGCAGAG
#&gt; colData names(9): log10FeaturesPerCount mitoRatio ... sampleID
#&gt;   sampleName
#&gt; reducedDimNames(0):
#&gt; altExpNames(0):</div><div class='input'>
<span class='va'>x</span> <span class='op'>&lt;-</span> <span class='fu'>bcbioSingleCell</span><span class='op'>(</span>
    uploadDir <span class='op'>=</span> <span class='va'>uploadDir</span>,
    sampleMetadataFile <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/file.path.html'>file.path</a></span><span class='op'>(</span><span class='va'>uploadDir</span>, <span class='st'>"metadata.csv"</span><span class='op'>)</span>
<span class='op'>)</span>
</div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>──</span><span> </span><span style='font-weight: bold;'>bcbioSingleCell</span><span> </span><span style='color: #00BBBB;'>──────────────────────────────────</span><span></span></div><div class='output co'>#&gt; <span class='message'>Importing bcbio-nextgen single-cell RNA-seq run</span></div><div class='output co'>#&gt; <span class='message'>projectDir: 2018-01-01_bcbio</span></div><div class='output co'>#&gt; <span class='message'>1 sample detected:</span></div><div class='output co'>#&gt; <span class='message'>    ● multiplexed_AAAAAAAA</span></div><div class='output co'>#&gt; <span class='message'>→ Importing </span><span style='color: #0000BB;'>project-summary.yaml</span><span> at </span><span style='color: #0000BB;'>/private/var/folders/l1/8y8sjzmn15v49jgrqglghcfr0000gn/T/Rtmpj5yP1c/temp_libpathe10850cb4eb4/bcbioSingleCell/extdata/indrops/2018-01-01_bcbio</span><span> using </span><span style='color: #0000BB;'>yaml</span><span>::</span><span style='color: #000000;background-color: #BBBBBB;'>`yaml.load_file()`</span><span>.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #BBBB00;'>!</span><span> Data versions are missing.</span></div><div class='output co'>#&gt; <span class='message'>→ Importing </span><span style='color: #0000BB;'>programs.txt</span><span> at </span><span style='color: #0000BB;'>/private/var/folders/l1/8y8sjzmn15v49jgrqglghcfr0000gn/T/Rtmpj5yP1c/temp_libpathe10850cb4eb4/bcbioSingleCell/extdata/indrops/2018-01-01_bcbio</span><span> using </span><span style='color: #0000BB;'>vroom</span><span>::</span><span style='color: #000000;background-color: #BBBBBB;'>`vroom()`</span><span>.</span></div><div class='output co'>#&gt; <span class='message'>→ Importing </span><span style='color: #0000BB;'>bcbio-nextgen.log</span><span> at </span><span style='color: #0000BB;'>/private/var/folders/l1/8y8sjzmn15v49jgrqglghcfr0000gn/T/Rtmpj5yP1c/temp_libpathe10850cb4eb4/bcbioSingleCell/extdata/indrops/2018-01-01_bcbio</span><span> using </span><span style='color: #0000BB;'>readr</span><span>::</span><span style='color: #000000;background-color: #BBBBBB;'>`read_lines()`</span><span>.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #BBBB00;'>!</span><span> </span><span style='color: #0000BB;'>bcbio-nextgen.log</span><span> file is empty.</span></div><div class='output co'>#&gt; <span class='message'>→ Importing </span><span style='color: #0000BB;'>bcbio-nextgen-commands.log</span><span> at </span><span style='color: #0000BB;'>/private/var/folders/l1/8y8sjzmn15v49jgrqglghcfr0000gn/T/Rtmpj5yP1c/temp_libpathe10850cb4eb4/bcbioSingleCell/extdata/indrops/2018-01-01_bcbio</span><span> using </span><span style='color: #0000BB;'>readr</span><span>::</span><span style='color: #000000;background-color: #BBBBBB;'>`read_lines()`</span><span>.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> 1000 reads per cellular barcode cutoff detected.</span></div><div class='output co'>#&gt; <span class='message'>Counts will imported as genes.</span></div><div class='output co'>#&gt; <span class='message'>UMI type: harvard-indrop-v3</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>── </span><span style='font-weight: bold;'>Sample metadata</span><span> ──</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>→ Importing </span><span style='color: #0000BB;'>metadata.csv</span><span> at </span><span style='color: #0000BB;'>/private/var/folders/l1/8y8sjzmn15v49jgrqglghcfr0000gn/T/Rtmpj5yP1c/temp_libpathe10850cb4eb4/bcbioSingleCell/extdata/indrops</span><span> using </span><span style='color: #0000BB;'>vroom</span><span>::</span><span style='color: #000000;background-color: #BBBBBB;'>`vroom()`</span><span>.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BBBB;'>ℹ</span><span> Multiplexed samples detected.</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>── </span><span style='font-weight: bold;'>Counts</span><span> ──</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>→ Importing counts.</span></div><div class='output co'>#&gt; <span class='message'>→ Importing </span><span style='color: #0000BB;'>multiplexed-AAAAAAAA.mtx.rownames</span><span> at </span><span style='color: #0000BB;'>/private/var/folders/l1/8y8sjzmn15v49jgrqglghcfr0000gn/T/Rtmpj5yP1c/temp_libpathe10850cb4eb4/bcbioSingleCell/extdata/indrops/multiplexed-AAAAAAAA</span><span> using </span><span style='color: #0000BB;'>readr</span><span>::</span><span style='color: #000000;background-color: #BBBBBB;'>`read_lines()`</span><span>.</span></div><div class='output co'>#&gt; <span class='message'>→ Importing </span><span style='color: #0000BB;'>multiplexed-AAAAAAAA.mtx.colnames</span><span> at </span><span style='color: #0000BB;'>/private/var/folders/l1/8y8sjzmn15v49jgrqglghcfr0000gn/T/Rtmpj5yP1c/temp_libpathe10850cb4eb4/bcbioSingleCell/extdata/indrops/multiplexed-AAAAAAAA</span><span> using </span><span style='color: #0000BB;'>readr</span><span>::</span><span style='color: #000000;background-color: #BBBBBB;'>`read_lines()`</span><span>.</span></div><div class='output co'>#&gt; <span class='message'>→ Importing </span><span style='color: #0000BB;'>multiplexed-AAAAAAAA.mtx</span><span> at </span><span style='color: #0000BB;'>/private/var/folders/l1/8y8sjzmn15v49jgrqglghcfr0000gn/T/Rtmpj5yP1c/temp_libpathe10850cb4eb4/bcbioSingleCell/extdata/indrops/multiplexed-AAAAAAAA</span><span> using </span><span style='color: #0000BB;'>Matrix</span><span>::</span><span style='color: #000000;background-color: #BBBBBB;'>`readMM()`</span><span>.</span></div><div class='output co'>#&gt; <span class='message'>→ Importing sidecar </span><span style='color: #0000BB;'>multiplexed-AAAAAAAA.mtx.rownames</span><span> at </span><span style='color: #0000BB;'>/private/var/folders/l1/8y8sjzmn15v49jgrqglghcfr0000gn/T/Rtmpj5yP1c/temp_libpathe10850cb4eb4/bcbioSingleCell/extdata/indrops/multiplexed-AAAAAAAA</span><span>.</span></div><div class='output co'>#&gt; <span class='message'>→ Importing </span><span style='color: #0000BB;'>multiplexed-AAAAAAAA.mtx.rownames</span><span> at </span><span style='color: #0000BB;'>/private/var/folders/l1/8y8sjzmn15v49jgrqglghcfr0000gn/T/Rtmpj5yP1c/temp_libpathe10850cb4eb4/bcbioSingleCell/extdata/indrops/multiplexed-AAAAAAAA</span><span> using </span><span style='color: #0000BB;'>readr</span><span>::</span><span style='color: #000000;background-color: #BBBBBB;'>`read_lines()`</span><span>.</span></div><div class='output co'>#&gt; <span class='message'>→ Importing sidecar </span><span style='color: #0000BB;'>multiplexed-AAAAAAAA.mtx.colnames</span><span> at </span><span style='color: #0000BB;'>/private/var/folders/l1/8y8sjzmn15v49jgrqglghcfr0000gn/T/Rtmpj5yP1c/temp_libpathe10850cb4eb4/bcbioSingleCell/extdata/indrops/multiplexed-AAAAAAAA</span><span>.</span></div><div class='output co'>#&gt; <span class='message'>→ Importing </span><span style='color: #0000BB;'>multiplexed-AAAAAAAA.mtx.colnames</span><span> at </span><span style='color: #0000BB;'>/private/var/folders/l1/8y8sjzmn15v49jgrqglghcfr0000gn/T/Rtmpj5yP1c/temp_libpathe10850cb4eb4/bcbioSingleCell/extdata/indrops/multiplexed-AAAAAAAA</span><span> using </span><span style='color: #0000BB;'>readr</span><span>::</span><span style='color: #000000;background-color: #BBBBBB;'>`read_lines()`</span><span>.</span></div><div class='output co'>#&gt; <span class='message'>→ Imported </span><span style='color: #0000BB;'>multiplexed-AAAAAAAA</span><span>.</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>── </span><span style='font-weight: bold;'>Feature metadata</span><span> ──</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>bcbio GTF file:</span>
#&gt; <span class='message'>  /n/app/bcbio/dev/genomes/Hsapiens/hg38/rnaseq/ref-transcripts.gtf</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #BBBB00;'>!</span><span> bcbio GTF file is not accessible.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #BBBB00;'>!</span><span> Slotting empty ranges into </span><span style='color: #000000;background-color: #BBBBBB;'>`rowRanges()`</span><span>.</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>── </span><span style='font-weight: bold;'>Column data</span><span> ──</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>── </span><span style='font-weight: bold;'>Metadata</span><span> ──</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>→ Importing unfiltered cellular barcode distributions.</span></div><div class='output co'>#&gt; <span class='message'>→ Importing </span><span style='color: #0000BB;'>multiplexed-AAAAAAAA-barcodes.tsv</span><span> at </span><span style='color: #0000BB;'>/private/var/folders/l1/8y8sjzmn15v49jgrqglghcfr0000gn/T/Rtmpj5yP1c/temp_libpathe10850cb4eb4/bcbioSingleCell/extdata/indrops/multiplexed-AAAAAAAA</span><span> using </span><span style='color: #0000BB;'>vroom</span><span>::</span><span style='color: #000000;background-color: #BBBBBB;'>`vroom()`</span><span>.</span></div><div class='output co'>#&gt; <span class='message'>→ Calculating 100 sample metrics.</span></div><div class='output co'>#&gt; <span class='message'></span><span style='color: #BBBB00;'>!</span><span> Calculating metrics without biotype information.</span>
#&gt; <span class='message'></span><span style='color: #000000;background-color: #BBBBBB;'>`rowRanges()`</span><span> required to calculate: </span><span style='color: #000000;background-color: #BBBBBB;'>`nCoding, nMito, mitoRatio`</span><span>.</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'></span><span style='color: #00BB00;'>✓</span><span> bcbio single-cell RNA-seq run imported successfully.</span></div><div class='input'><span class='fu'><a href='https://rdrr.io/r/base/print.html'>print</a></span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span>
</div><div class='output co'>#&gt; bcbioSingleCell 0.4.13
#&gt; uploadDir: /private/var/folders/l1/8y8sjzmn15v49jgrqglghcfr0000gn/T/Rtmpj5yP1c/temp_libpathe10850cb4eb4/bcbioSingleCell/extdata/indrops
#&gt; dates(2): [bcbio] 2018-01-01; [R] 2020-10-08
#&gt; level: genes
#&gt; sampleMetadataFile: /private/var/folders/l1/8y8sjzmn15v49jgrqglghcfr0000gn/T/Rtmpj5yP1c/temp_libpathe10850cb4eb4/bcbioSingleCell/extdata/indrops/metadata.csv
#&gt; interestingGroups: sampleName
#&gt; filtered: FALSE
#&gt; class: SingleCellExperiment 
#&gt; dim: 50 100 
#&gt; metadata(27): allSamples bcbioCommandsLog ... wd yaml
#&gt; assays(1): counts
#&gt; rownames(50): ENSG00000071082 ENSG00000100316 ... ENSG00000269028
#&gt;   ENSG00000282105
#&gt; rowData names(0):
#&gt; colnames(100): AAACACTA_CTTCGATT AAACTACA_CCACATTA ...
#&gt;   TGGGAATT_ATATAGGA TGTTATCA_ACGCAGAG
#&gt; colData names(15): aggregate description ... sampleName sequence
#&gt; reducedDimNames(0):
#&gt; altExpNames(0):</div></span></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by <a href='https://steinbaugh.com/'>Michael Steinbaugh</a>, <a href='https://github.com/roryk/'>Rory Kirchner</a>, <a href='https://github.com/marypiper/'>Mary Piper</a>, <a href='https://github.com/vbarrera/'>Victor Barrera</a>, <a href='https://github.com/sjhosui/'>Shannan Ho Sui</a>, <a href='https://bioinformatics.sph.harvard.edu/'>Harvard Chan Bioinformatics Core</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


