<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Extract or replace parts of an object — extract • bcbioSingleCell</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



  <link href="../extra.css" rel="stylesheet">
  

<meta property="og:title" content="Extract or replace parts of an object — extract" />
<meta property="og:description" content="Extract genes by row and cells by column." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">bcbioSingleCell</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.4.13</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/hbc/bcbioSingleCell/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Extract or replace parts of an object</h1>
    <small class="dont-index">Source: <a href='https://github.com/hbc/bcbioSingleCell/blob/master/R/extract-methods.R'><code>R/extract-methods.R</code></a></small>
    <div class="hidden name"><code>extract.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Extract genes by row and cells by column.</p>
    </div>

    <pre class="usage"># S4 method for bcbioSingleCell,ANY,ANY,ANY
[(x, i, j, ..., drop = FALSE)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>x</th>
      <td><p>object from which to extract element(s) or in which to replace element(s).</p></td>
    </tr>
    <tr>
      <th>i</th>
      <td><p>indices specifying elements to extract or replace.  Indices are
    <code>numeric</code> or <code>character</code> vectors or empty (missing) or
    <code>NULL</code>.  Numeric values are coerced to integer as by
    <code><a href='https://rdrr.io/r/base/integer.html'>as.integer</a></code> (and hence truncated towards zero).
    Character vectors will be matched to the <code><a href='https://rdrr.io/r/base/names.html'>names</a></code> of the
    object (or for matrices/arrays, the <code><a href='https://rdrr.io/r/base/dimnames.html'>dimnames</a></code>):
    see &#8216;Character indices&#8217; below for further details.</p>
<p>For <code><a href='https://rdrr.io/r/base/Extract.html'>[</a></code>-indexing only: <code>i</code>, <code>j</code>, <code>...</code> can be
    logical vectors, indicating elements/slices to select.  Such vectors
    are recycled if necessary to match the corresponding extent.
    <code>i</code>, <code>j</code>, <code>...</code> can also be negative integers,
    indicating elements/slices to leave out of the selection.</p>
<p>When indexing arrays by <code><a href='https://rdrr.io/r/base/Extract.html'>[</a></code> a single argument <code>i</code> can be a
    matrix with as many columns as there are dimensions of <code>x</code>; the
    result is then a vector with elements corresponding to the sets of
    indices in each row of <code>i</code>.</p>
<p>An index value of <code>NULL</code> is treated as if it were <code><a href='https://rdrr.io/r/base/integer.html'>integer(0)</a></code>.</p></td>
    </tr>
    <tr>
      <th>j</th>
      <td><p>indices specifying elements to extract or replace.  Indices are
    <code>numeric</code> or <code>character</code> vectors or empty (missing) or
    <code>NULL</code>.  Numeric values are coerced to integer as by
    <code><a href='https://rdrr.io/r/base/integer.html'>as.integer</a></code> (and hence truncated towards zero).
    Character vectors will be matched to the <code><a href='https://rdrr.io/r/base/names.html'>names</a></code> of the
    object (or for matrices/arrays, the <code><a href='https://rdrr.io/r/base/dimnames.html'>dimnames</a></code>):
    see &#8216;Character indices&#8217; below for further details.</p>
<p>For <code><a href='https://rdrr.io/r/base/Extract.html'>[</a></code>-indexing only: <code>i</code>, <code>j</code>, <code>...</code> can be
    logical vectors, indicating elements/slices to select.  Such vectors
    are recycled if necessary to match the corresponding extent.
    <code>i</code>, <code>j</code>, <code>...</code> can also be negative integers,
    indicating elements/slices to leave out of the selection.</p>
<p>When indexing arrays by <code><a href='https://rdrr.io/r/base/Extract.html'>[</a></code> a single argument <code>i</code> can be a
    matrix with as many columns as there are dimensions of <code>x</code>; the
    result is then a vector with elements corresponding to the sets of
    indices in each row of <code>i</code>.</p>
<p>An index value of <code>NULL</code> is treated as if it were <code><a href='https://rdrr.io/r/base/integer.html'>integer(0)</a></code>.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Additional arguments.</p></td>
    </tr>
    <tr>
      <th>drop</th>
      <td><p>For matrices and arrays.  If <code>TRUE</code> the result is
    coerced to the lowest possible dimension (see the examples).  This
    only works for extracting elements, not for the replacement.  See
    <code><a href='https://rdrr.io/r/base/drop.html'>drop</a></code> for further details.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p><code>bcbioSingleCell</code>.</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>Refer to <code><a href='https://basejump.acidgenomics.com/reference/cell2sample.html'>cell2sample()</a></code> and
<a href='https://basejump.acidgenomics.com/reference/selectSamples.html'>selectSamples()</a> if sample-level extraction is
desired. Note that <code>sampleID</code> is slotted into <code>colData</code> and defines the
cell-to-sample mappings.</p>
<p>Unfiltered cellular barcode distributions for the entire dataset, including
cells not kept in the matrix will be dropped in favor of the <code>nCount</code> column
of <code><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>colData()</a></code>.</p>
    <h2 class="hasAnchor" id="note"><a class="anchor" href="#note"></a>Note</h2>

    <p>Updated 2019-08-20.</p>
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Becker, R. A., Chambers, J. M. and Wilks, A. R. (1988)
  <em>The New S Language</em>.
  Wadsworth &amp; Brooks/Cole.</p>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Michael Steinbaugh</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='va'>bcb</span><span class='op'>)</span>

<span class='va'>cells</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>bcb</span><span class='op'>)</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span><span class='op'>(</span><span class='va'>cells</span><span class='op'>)</span>
</div><div class='output co'>#&gt; [1] "AAACACTA_CTTCGATT" "AAACTACA_CCACATTA" "AACTGCCT_GCAAGGAC"
#&gt; [4] "AAGAAGGT_TCTGTGGT" "AAGCCTTC_TAAATAGG" "AATAAGGA_CCACATTA"</div><div class='input'><span class='va'>genes</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>bcb</span><span class='op'>)</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span><span class='op'>(</span><span class='va'>genes</span><span class='op'>)</span>
</div><div class='output co'>#&gt; [1] "ENSG00000071082" "ENSG00000100316" "ENSG00000106631" "ENSG00000108821"
#&gt; [5] "ENSG00000112306" "ENSG00000115414"</div><div class='input'>
<span class='co'>## Subset by cell identifiers.</span>
<span class='va'>bcb</span><span class='op'>[</span>, <span class='va'>cells</span><span class='op'>]</span>
</div><div class='output co'>#&gt; bcbioSingleCell 0.3.19
#&gt; uploadDir: /data00/draco/acidbase/packages/bcbioSingleCell/inst/extdata/indrops
#&gt; dates(2): [bcbio] 2018-01-01; [R] 2019-08-12
#&gt; level: genes
#&gt; sampleMetadataFile: inst/extdata/indrops/metadata.csv
#&gt; organism: Homo sapiens
#&gt; interestingGroups: sampleName
#&gt; filtered: FALSE
#&gt; class: SingleCellExperiment 
#&gt; dim: 50 6 
#&gt; metadata(27): allSamples bcbioCommandsLog ... yaml subset
#&gt; assays(1): counts
#&gt; rownames(50): ENSG00000071082 ENSG00000100316 ... ENSG00000269028
#&gt;   ENSG00000282105
#&gt; rowData names(8): broadClass description ... geneName seqCoordSystem
#&gt; colnames(6): AAACACTA_CTTCGATT AAACTACA_CCACATTA ... AAGCCTTC_TAAATAGG
#&gt;   AATAAGGA_CCACATTA
#&gt; colData names(15): aggregate description ... sampleName sequence
#&gt; reducedDimNames(0):
#&gt; altExpNames(0):</div><div class='input'>
<span class='co'>## Subset by genes.</span>
<span class='va'>bcb</span><span class='op'>[</span><span class='va'>genes</span>, <span class='op'>]</span>
</div><div class='output co'>#&gt; bcbioSingleCell 0.3.19
#&gt; uploadDir: /data00/draco/acidbase/packages/bcbioSingleCell/inst/extdata/indrops
#&gt; dates(2): [bcbio] 2018-01-01; [R] 2019-08-12
#&gt; level: genes
#&gt; sampleMetadataFile: inst/extdata/indrops/metadata.csv
#&gt; organism: Homo sapiens
#&gt; interestingGroups: sampleName
#&gt; filtered: FALSE
#&gt; class: SingleCellExperiment 
#&gt; dim: 6 100 
#&gt; metadata(27): allSamples bcbioCommandsLog ... yaml subset
#&gt; assays(1): counts
#&gt; rownames(6): ENSG00000071082 ENSG00000100316 ... ENSG00000112306
#&gt;   ENSG00000115414
#&gt; rowData names(8): broadClass description ... geneName seqCoordSystem
#&gt; colnames(100): AAACACTA_CTTCGATT AAACTACA_CCACATTA ...
#&gt;   TGGGAATT_ATATAGGA TGTTATCA_ACGCAGAG
#&gt; colData names(15): aggregate description ... sampleName sequence
#&gt; reducedDimNames(0):
#&gt; altExpNames(0):</div><div class='input'>
<span class='co'>## Subset by both genes and cells.</span>
<span class='va'>bcb</span><span class='op'>[</span><span class='va'>genes</span>, <span class='va'>cells</span><span class='op'>]</span>
</div><div class='output co'>#&gt; bcbioSingleCell 0.3.19
#&gt; uploadDir: /data00/draco/acidbase/packages/bcbioSingleCell/inst/extdata/indrops
#&gt; dates(2): [bcbio] 2018-01-01; [R] 2019-08-12
#&gt; level: genes
#&gt; sampleMetadataFile: inst/extdata/indrops/metadata.csv
#&gt; organism: Homo sapiens
#&gt; interestingGroups: sampleName
#&gt; filtered: FALSE
#&gt; class: SingleCellExperiment 
#&gt; dim: 6 6 
#&gt; metadata(27): allSamples bcbioCommandsLog ... yaml subset
#&gt; assays(1): counts
#&gt; rownames(6): ENSG00000071082 ENSG00000100316 ... ENSG00000112306
#&gt;   ENSG00000115414
#&gt; rowData names(8): broadClass description ... geneName seqCoordSystem
#&gt; colnames(6): AAACACTA_CTTCGATT AAACTACA_CCACATTA ... AAGCCTTC_TAAATAGG
#&gt;   AATAAGGA_CCACATTA
#&gt; colData names(15): aggregate description ... sampleName sequence
#&gt; reducedDimNames(0):
#&gt; altExpNames(0):</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by <a href='https://steinbaugh.com/'>Michael Steinbaugh</a>, <a href='https://github.com/roryk/'>Rory Kirchner</a>, <a href='https://github.com/marypiper/'>Mary Piper</a>, <a href='https://github.com/vbarrera/'>Victor Barrera</a>, <a href='https://github.com/sjhosui/'>Shannan Ho Sui</a>, <a href='https://bioinformatics.sph.harvard.edu/'>Harvard Chan Bioinformatics Core</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


